### PR TITLE
Add OpsGenie pack

### PR DESCRIPTION
**Pull request by jjm**
### Status 

- new pack.

### Description

Add new pack for the OpsGenie Service.

### Checklist (tick everything that applies)

- [X] Metadata: pack.yaml, icon, structure (required for new packs)
- [X] Version bump (required for changed packs)
- [X] Code linting (required, can be done after the PR checks)
- [X] [Tests](https://docs.stackstorm.com/development/pack_testing.html) (not required but really recommended)
- [x] Install from GitHub via `pack deploy`.

### CODE REVIEW

**Comment by jjm** [comment]
I just need to review and check I've not missed any useful API calls.

The ToDo list:

[ ] Add docstrings.
[ ] Check test coverage.
[ ] Review `Create Alert` as this is the most complex API call with the most options.

File: packs/opsgenie/README.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# OpsGenie Pack
+
+To use this intergration pack, you'll need an OpsGenie account and an
```
Comment (armab) [inline packs/opsgenie/README.md]: Some two-words description about `opsgenie` would be good, or at least a link.
My thinking is: there are a lot of services and sometimes it's hard to know without googling.

Here is an example with brief description: https://github.com/StackStorm/st2contrib/blob/master/packs/yammer/README.md#yammer-integration-pack
^^ I think that shows good practice.

Comment (LindsayHill) [inline packs/opsgenie/README.md]: s/intergration/integration/

**Comment by Kami** [comment]
Awesome, nice work!

File: packs/opsgenie/README.md (hunk 1 above)
Comment (Kami) [inline packs/opsgenie/README.md]: I agree with @armab, should describe what the service does would be great.

Comment (jjm) [inline packs/opsgenie/README.md]: I've a good suggestion, I'll do along with additions I've got planned.

File: packs/opsgenie/actions/add_heartbeat.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+from lib.actions import OpsGenieBaseAction
+
+
+class AddHeartbeatAction(OpsGenieBaseAction):
+    def run(self, name, interval=None, intervalUnit=None, description=None, enabled=False):
+        """
```
Comment (lakshmi-kannan) [inline packs/opsgenie/actions/add_heartbeat.py]: Goober. Please remove.

File: packs/opsgenie/actions/add_heartbeat.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+parameters:
+  name:
+    type: "string"
+    description: "Name of the heartbeat."
+    required: true
+  interval:
+    type: "integer"
+    description: "Specifies how often a heartbeat message should be expected."
+    default: ~
```
Comment (lakshmi-kannan) [inline packs/opsgenie/actions/add_heartbeat.yaml]: Does this even work?

Comment (DoriftoShoes) [inline packs/opsgenie/actions/add_heartbeat.yaml]: `~` means null in yaml.

File: packs/opsgenie/actions/add_heartbeat.py (hunk 2 above)
Comment (lakshmi-kannan) [inline packs/opsgenie/actions/add_heartbeat.py]: NVM, see your TODO!

File: packs/opsgenie/actions/add_heartbeat.yaml (hunk 3 above)
Comment (lakshmi-kannan) [inline packs/opsgenie/actions/add_heartbeat.yaml]: Ahh. So much magic.

File: packs/opsgenie/actions/add_heartbeat.py (hunk 2 above)
Comment (jjm) [inline packs/opsgenie/actions/add_heartbeat.py]: No worries, I'm working my way though adding docstrings and they should all be covered in he next commit.

File: packs/opsgenie/actions/add_heartbeat.yaml (hunk 3 above)
Comment (jjm) [inline packs/opsgenie/actions/add_heartbeat.yaml]: It sure does work and the README has an example of using this action to setup a heartbeat between 2 ST2 instances to check ChatOps is functional.

File: packs/opsgenie/aliases/get_schedule_who_is_on_call.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (emedvedev) [inline packs/opsgenie/aliases/get_schedule_who_is_on_call.yaml]: :heart:

File: packs/opsgenie/aliases/get_schedule_who_is_on_call.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  format: |
+    {% if execution.status == 'succeeded' %}
+    On call:{~}
+    {% for item in execution.result.result.oncalls %}
+    {{item.name}}: {% for oncall in item.participants %}{{oncall.name}} {% endfor %}
+
+    {% endfor %}
+    {% else %}
+    Could not query OpsGenie. See see {{execution.id}} for details.{~}
+    {% endif %}
```
Comment (emedvedev) [inline packs/opsgenie/aliases/get_schedule_who_is_on_call.yaml]: :heart:

Comment (jjm) [inline packs/opsgenie/aliases/get_schedule_who_is_on_call.yaml]: I like adding aliases :-D

**Comment by jjm** [comment]
I've completed my 3 todo list items and the example in the README has been running fine for 24 hours. So I think it's good to merge if there is not any more review comments.

I'm not sure why the build failed, it's passing locally.

**Comment by Kami** [comment]
Merged, thanks.

Similar to other PR I will try to resolve travis issues on master.

**Pull request merged**